weave-js/src/components/PagePanel.tsx

```diff
@@ -110,6 +112,77 @@ const JupyterControlsIcon = styled.div`
   }
 `;
 
+const HOST_SESSION_ID_COOKIE = `host_session_id`;
+
+function useEnablePageAnalytics() {
+  const history = useHistory();
+  const pathRef = useRef('');
+  const {urlPrefixed, backendWeaveViewerUrl} = getConfig();
+  const [userId, setUserId] = useState('');
+
+  // fetch user
+  useEffect(() => {
+    const anonApiKey = getCookie('anon_api_key');
+    const additionalHeaders: Record<string, string> = {};
+    if (anonApiKey != null && anonApiKey !== '') {
+      additionalHeaders['x-wandb-anonymous-auth-id'] = btoa(anonApiKey);
+    }
+    fetch(urlPrefixed(backendWeaveViewerUrl()), {
+      credentials: 'include',
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        ...additionalHeaders,
+      },
+    })
+      .then(res => {
+        if (res.status === 200) {
+          return res.json();
+        }
+        return;
+      })
+      .then(json => {
+        const serverUserId = json?.user_id ?? '';
+        if (serverUserId !== userId) {
+          setUserId(serverUserId);
```

Comment:
This is going to create a double request for every page load since the use effect is dependent on user id. It would be better to use the `userId` in `setUserId`'s callback

Reply:
Honestly, I can just remove the state since we're not returning it. And just call identify when we get back a serverUserId. And then remove userId from the dependency list.

Reply:
added in next commit